src/background/Program.ts

```diff
@@ -2083,11 +2087,23 @@ export default class BackgroundProgram {
       raw: rawOptions,
       errors: [],
       mac,
+      chromiumVariant,
     };
 
     log.level = options.logLevel;
   }
 
+  async getChromiumVariant(): Promise<string> {
+    if (BROWSER !== "chrome") {
+      return "";
+    }
+    const tabs = await browser.tabs.query({
+      active: true,
+      currentWindow: true,
+    });
+    return tabs?.[0]?.vivExtData !== undefined ? "vivaldi" : "chrome";
+  }
```

Comment:
Could you move this to a function at the bottom of the file since it never uses `this`?

Would it make sense to run this _every time_ inside the `openNewTab` method, instead of just once at startup basically? I think it should be so fast that it won’t be noticeable. The reason is that it would feel nice to have the Vivaldi stuff localized to the only place where it matters, instead of having `this.chromiumVariant`. If we ever need Vivaldi specific stuff in the future, we can refactor then. You can use the `fireAndForget` helper (which is already used in the method) to avoid `openNewTab` becoming `async`.

Reply:
> Could you move this to a function at the bottom of the file since it never uses `this`?

Definitely.




> Would it make sense to run this _every time_ inside the `openNewTab` method, instead of just once at startup basically?

I totally appreciate the desire to keep the necessary changes localized, and I wouldn't mind accommodating that, even though I do think it makes more sense to establish the specific browser once (since it won't be changing) and passing it around where it's needed, but I think the complication with the tab API being asynchronous is a little more sticky unless I'm misunderstanding something.

> You can use the `fireAndForget` helper (which is already used in the method) to avoid `openNewTab` becoming `async`.

I don't see how that will work since we actually need the value wrapped by the promise returned by `browser.tabs.query()`. It looks to me like `fireAndForget()` would only help if we didn't care about resolving the promise. I don't see a way to use the result of `browser.tabs.query()` without making the logic downstream of that also asynchronous/promise-aware. That struck me as a bigger change with more potential implication, which is why I opted to do it in the background program initialization, which seemed to already have other asynchronous work.

If you'd strongly prefer to do the browser detection right before the click, I can try it out.